source/features/fork-source-link-same-view.tsx

```diff
@@ -0,0 +1,66 @@
+import select from 'select-dom';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+import * as api from '../github-helpers/api';
+import GitHubURL from '../github-helpers/github-url';
+import {getRepoURL} from '../github-helpers';
+
+const checkIfFileExists = async (url: GitHubURL) => {
+	const {repository} = await api.v4(`
+		repository(owner: "${url.user}", name: "${url.repository}") {
+			file: object(expression: "${url.branch}:${url.filePath}") {
+				... on Blob {
+					id
+				}
+			}
+		}
+	`);
+
+	if (!repository.file) {
+		const source = select<HTMLAnchorElement>(`[href$="${url.pathname}" i]`)!;
+		source.pathname = '/' + source.textContent!.trim();
+	}
+};
+
+// eslint-disable-next-line import/prefer-default-export
+export function createHEADLink(baseRepo: string): string {
+	if (pageDetect.isRepoRoot() || !(pageDetect.isSingleFile() || pageDetect.isRepoTree() || // There will be no change to the link
+		baseRepo.toLowerCase() === getRepoURL()) // Your there already
```

Comment:
This line will happen when there are multiple forks.

Reply:
??

The link we're changing always points to **one** repo. Also checks your "you're", it's typoed in every JS comment 😁